omnigibson/utils/registry_utils.py

```diff
@@ -336,10 +332,8 @@ def _dump_state(self):
 
     def _load_state(self, state):
         # Iterate over all objects and load their states
+        assert {obj.name for obj in self.objects} == state.keys(), "object mismatch when loading state for registry"
```

Comment:
does this actually work? doesn't state.keys() return a generator object, not a set?

Reply:
You are right. But it actually works. I guess python is smart about it.